docs/100_OVERVIEW.md

```diff
@@ -0,0 +1,147 @@
+
+### Plugin
+The `Plugin` provides methods for LibOCR to call as a valid OCR2 plugin. The 
+responsibility of each method is defined below. Generally, a `Plugin` gets
+observations from a list of multiple `Observer`, uses a `Merger` to merge
+observations from the network, and an `Encoder` to produce a transmittable
+report.
+
+#### Rules
+- A `Plugin` only interacts with `Observer`, `Merger`, and `Encoder`
+- A `Plugin` should not know anything about the structural contents of a `BlockKey`
+- A `Plugin` should not know anything about the structural contents of a `UpkeepIdentifier`
+- A `Plugin` should start and stop services per its appropriate life cycle
+- A `Plugin` should treat all incoming observations as potentially malicious
+- A `Plugin` should isolate all state interactions to instances of `Observer`
+
+#### Observation
+The `Observation` method should loop through multiple `Observer` instances to
+collect pending and eligible upkeep results. Those results should be packaged 
+together as an automation observation (distinct type) and byte encoded. This 
+function should never directly interact with the encoded upkeep results. Its 
+sole responsibilities should remain as a collector and basic encoder to the 
+final transport type.
+
+More about specifics of an observation from an observer here:
+[OBSERVATION](./300_OBSERVATION.md)
+
+##### Observation Struct
+An observation is composed of a list of eligible upkeeps and a list of pending
+upkeeps. Providing both allows for all nodes to produce a report without
+checking internal state. Versioning allows for version checks to be applied as
+new nodes can join the network with different active versions.
+
+```
+type VersionedObservation struct {
+    // Version is intended to allow for observations to change over time
+    Version     string
+    // Encoding indicates the type of encoder/registry used to build the
+    // included observations
+    Encoding    string
+}
+
+// ObservationV10 is the intended first step in observation versioning
+type ObservationV10 struct {
+    VersionedObservation
+    // Block contains the block value for nodes to coordinate on
+    Block BlockKey
+    // Data contains actual values of this version of observation
+    Data []UpkeepIdentifier
+}
+
+// ObservationV11 displays a potential way to differentiate observation versions
+type ObservationV11 struct {
+    VersionedObservation
+    // Block contains the block value for nodes to coordinate on
+    Block BlockKey
+    // Data contains actual values of this version of observation
+    Data       []UpkeepIdentifier
+    // OtherStuff in this case is addative and the version minor is incremented
+    OtherStuff int
+}
+
+// ObservationV20 displays a potential way to differentiate observation versions
+type ObservationV20 struct {
+    VersionedObservation
+    // Data in this version is a different structure and would break 
+    // compatibility to previous versions
+    Data       map[int]PointState
```

Comment:
curious why you envision a map of point states here? this is only a single observation so cannot compute a point state yet?

Reply:
This is a future concept. A `PointState` is part of a re-architecture, so that can be ignored for now. The premise here is that a VersionedObservation can have different data structures per version.